### PR TITLE
Intel HPC Platform Spec Integration Tests

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/intel_hpc/test_intel_hpc.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+import re
+
+import pytest
+
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from tests.common.assertions import assert_no_errors_in_logs
+from tests.common.mpi_common import _test_mpi
```
Comment (enrico-usai) [inline tests/integration-tests/tests/intel_hpc/test_intel_hpc.py]: you are importing a "private" function, we should remove the underscore.

File: tests/integration-tests/tests/intel_hpc/test_intel_hpc.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+from tests.common.schedulers_common import get_scheduler_commands
+from tests.common.utils import _fetch_instance_slots
```
Comment (enrico-usai) [inline tests/integration-tests/tests/intel_hpc/test_intel_hpc.py]: same as before.

File: tests/integration-tests/tests/intel_hpc/test_intel_hpc.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    remote_command_executor.run_remote_command(
+        "sudo cp ~/clck.xml /opt/intel/clck/2019.3.5/etc/clck.xml", additional_files=[str(test_datadir / "clck.xml")]
+    )
+
+    # Run Cluster Checker
+    result = remote_command_executor.run_remote_script(str(test_datadir / "run_clck.sh"))
+    try:
+        assert_that(result.stdout).contains("Overall Result: PASS")
+    except AssertionError as e:
+        logging.error(remote_command_executor.run_remote_command("cat clck_results.log"))
```
Comment (demartinofra) [inline tests/integration-tests/tests/intel_hpc/test_intel_hpc.py]: Probably a .stdout is missing after run_remote_command

File: tests/integration-tests/tests/intel_hpc/test_intel_hpc.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        raise (e)
```
Comment (demartinofra) [inline tests/integration-tests/tests/intel_hpc/test_intel_hpc.py]: minor: you can simply have `raise`

Comment (sean-smith) [reply tests/integration-tests/tests/intel_hpc/test_intel_hpc.py]: changed

File: tests/integration-tests/tests/intel_hpc/test_intel_hpc.py (hunk 3 above)
Comment (sean-smith) [reply tests/integration-tests/tests/intel_hpc/test_intel_hpc.py]: hmm... it works this way, but I've added it anyways

**Pull request merged**